CMakeLists.txt

```diff
@@ -2024,8 +2031,8 @@ if(NOT CURL_DISABLE_INSTALL)
   set(CC                      "${CMAKE_C_COMPILER}")
   # TODO: probably put a -D... options here?
   set(CONFIGURE_OPTIONS       "")
-  set(CURLVERSION             "${CURL_VERSION}")
-  set(VERSIONNUM              "${CURL_VERSION_NUM}")
+  set(CURLVERSION             "${LIBCURL_VERSION}")
+  set(VERSIONNUM              "${LIBCURL_VERSION_NUM}")
```

Comment:
What do you think of doing any renaming in a separate PR?

Reply:
I've separated it into a commit. If that's not enough I can create a new PR.

Reply:
I'd suggest a separate PR.

Reply:
`CURL_VERSION` will be initialized by `project()` (without the `-DEV` as suffix because the syntax rejects). The variables related to `project()` should be generated by `project()` IMO, though I'm not insist on it.

Reply:
Oh, so `project()` is setting the `CURL_VERSION` and `*_MAJOR/MINOR/PATCH` variables automatically, but also differently depending on version and/or CMP policy. E.g. cmake 3.30.x is setting them.

Before this patch we already set `CURL_VERSION` to `8.11.0-DEV`. This changes to `8.11.0` when using `project()`.

What is the advantage of using `project()` for setting the `CURL_VERSION` variable? Why do we need this?

Reply:
CPack will use some properties from `project()` (e.g. version) defaultly. I add the version for `project()` just because cpack generated a package with a meaningless version of `0.1.1`. 

Of course, we could set `CPACK_PACKAGE_VERSION_*` explicitly. But leaving a standard variable (standard means generated by standard way) is NOT BAD as many projects did. e.g. Besides <Project>_VERSION`, `PROJECT_VERSION` will be available in current scope, which sometimes helps to reuse some scripts more easily.

The original `CURL_VERSION` is merely an internally-used variable. Renaming it (to a still meaningful one as same as it's in `curlver.h`) seems no problem here.

As I've mentioned, I don't insist on the change. IMHO it's just not a complicated modification and can be improved here.

Reply:
Thanks for your insight. Agreed to make our existing variables "internal", meaning renaming them to the underscore-lowercase naming scheme already used for this (`_curl_version*`). These variables are only used inside the root `./CMakeLists.txt`. (And remain accessible by the rest if it becomes necessary.)

Existing `CURL_VERSION` could in theory be used externally, but it seems fine, as it isn't an interface with any promises. If anything it will be more alike other projects with this change.

My remaining concern is that the `-DEV` part of the curl version is actually significant, e.g. `8.11.0-DEV` is a different version than `8.11.0` stable. Publishing the version or a package without the `-DEV` distinction can be a source of misunderstandings.

Some projects solve this by using a distinctive, but numeric version, e.g. `8.10.90` (instead of 8.11.0-DEV). But that'd be a pretty far-reaching change in curl.

/cc @bagder

Reply:
...also, existing MAJOR/MINOR/PATCH and NUM version number already miss the `-DEV` distinction, which suggests we will be fine with this.

Reply:
> My remaining concern is that the `-DEV` part of the curl version is actually significant, e.g. `8.11.0-DEV` is a different version than `8.11.0` stable. Publishing the version or a package without the `-DEV` distinction can be a source of misunderstandings.

Surely, this is an important concern. Seems a defect by CMake for me. CMake provides the fourth part of version, [TWEAK](https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION_TWEAK.html), and I'm confused why it rejects the non-number style..

And also, CPack lacks the `TWEAK` part... But it seems there's a workaround:
```
string(REGEX REPLACE "([0-9]+\.[0-9]+)\.([0-9]+.+)" "\\2" LIBCURL_VERSION_PATCH_TWEAK "${LIBCURL_VERSION}")
set(CPACK_PACKAGE_VERSION_PATCH ${LIBCURL_VERSION_PATCH_TWEAK})
```

Reply:
Nice catch. As weird as it is, I think we can go with this.

perhaps also reduce to a one-liner:
```cmake
string(REGEX REPLACE "([0-9]+\.[0-9]+)\.([0-9]+.+)" "\\2" CPACK_PACKAGE_VERSION_PATCH "${LIBCURL_VERSION}")
```

wouldn't we need this to catch non-DEV versions?:
```cmake
string(REGEX REPLACE "([0-9]+\.[0-9]+)\.([0-9]+.*)" "\\2" CPACK_PACKAGE_VERSION_PATCH "${LIBCURL_VERSION}")
```

Reply:
> wouldn't we need this to catch non-DEV versions?:
> 
> ```cmake
> string(REGEX REPLACE "([0-9]+\.[0-9]+)\.([0-9]+.*)" "\\2" CPACK_PACKAGE_VERSION_PATCH "${LIBCURL_VERSION}")
> ```

Yes. I've tested both DEV and non-DEV versions. This works fine.

Reply:
Would you want to do the renaming from  `LIBCURL_VERSION*` to `_cmake_version*`?

Reply:
`CMAKE_VERSION` is the version of CMake. But we could add an underscore prefix (with full lower-case)?

Reply:
I meant rename to `_curl_version_*`, sorry.

Reply:
> I meant rename to `_curl_version_*`, sorry.

done